pkg/controller/customdomain/customdomain_controller_test.go

```diff
@@ -421,11 +423,38 @@ func TestCustomDomainController(t *testing.T) {
 	if err != nil {
 		t.Fatalf("get custom domain: (%v)", err)
 	}
+
 	// check for ready status
 	if actualCustomDomain.Status.State != customdomainv1alpha1.CustomDomainStateReady {
 		t.Errorf(fmt.Sprintf("Status.State does not equal (%s)", string(customdomainv1alpha1.CustomDomainStateReady)))
 	}
 
+	// Check scope immutability
+	externalScopePatchData := []byte(`{"spec":{"scope":"External"}}`)
+	err = r.client.Patch(context.TODO(), &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: instanceName,
+			Namespace: instanceNamespace,
+		},
+	}, client.ConstantPatch(types.StrategicMergePatchType, externalScopePatchData))
+	if err != nil {
+		t.Error("Unable to patch customdomain scope")
+	}
+
+	res, err = r.Reconcile(req)
+	if err == nil {
+		t.Error("Expected error when modifying Spec.Scope")
+	}
+
+	// Reset scope after testing
+	internalScopePatchData := []byte(`{"spec":{"scope":"Internal"}}`)
+	err = r.client.Patch(context.TODO(), &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: instanceName,
+			Namespace: instanceNamespace,
+		},
+	}, client.ConstantPatch(types.StrategicMergePatchType, internalScopePatchData))
+
```

Comment:
Looks like you need to check err here:
```suggestion
	if err != nil {
		t.Error("Unable to patch customdomain scope")
	}
```